packages/flutter_hooks/lib/src/debounced.dart

```diff
@@ -0,0 +1,74 @@
+part of 'hooks.dart';
+
+/// Returns a [ValueNotifier] that updates its value with a timeout
+/// for the given input [toDebounce] and triggers widget rebuilds accordingly.
+ValueNotifier<T?> useDebounced<T>(
+  T toDebounce, {
+  Duration timeout = const Duration(milliseconds: 500),
+}) {
+  return use(
+    _DebouncedHook(
+      toDebounce: toDebounce,
+      timeout: timeout,
+    ),
+  );
+}
+
+class _DebouncedHook<T> extends Hook<ValueNotifier<T?>> {
```

Comment:
Also, why are we using T? instead of T?

Reply:
If useDebounced is called for the first time and it's timer is still active, does that imply that there's no value yet (nothing debounced yet), making it null?